superbench/benchmarks/micro_benchmarks/memory_bw_performance_base.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""Module of the memory performance benchmarks base class."""
+
+from superbench.common.utils import logger
+from superbench.benchmarks import ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class MemBwBenchmark(MicroBenchmarkWithInvoke):
+    """The Cuda memory performance benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._mem_types = ['htod', 'dtoh', 'dtod']
+        self._metrics = ['H2D_Mem_BW', 'D2H_Mem_BW', 'D2D_Mem_BW']
+        self._memory = ['pinned', 'unpinned']
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+        self._parser.add_argument(
+            '--mem_type',
+            type=str,
+            nargs='+',
+            default=self._mem_types,
+            help='Memory types to benchmark. E.g. {}.'.format(' '.join(self._mem_types)),
+        )
+        self._parser.add_argument(
+            '--memory',
+            type=str,
+            default='pinned',
+            help='Memory argument for bandwidthtest. E.g. {}.'.format(' '.join(self._memory)),
+        )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        # Format the arguments
+        self._args.mem_type = [p.lower() for p in self._args.mem_type]
```

Comment:
Previously, we do the check ` if not isinstance(self._args.mem_type, list):`. Why do we remove this in the new version?

Reply:
Restored, but In fact, argparse will guarantee it's a list even user inputs a single one, so the place that may go wrong is the value assignment in the code, for example, default value assignment.

Reply:
We can remove it directly.